File: src/main/java/com/aml/trustwatch/service/TransactionService.java
package com.aml.trustwatch.service;

import com.aml.trustwatch.exception.ResourceNotFoundException;
import com.aml.trustwatch.loader.TransactionLoader;
import com.aml.trustwatch.model.Transaction;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class TransactionService {

    private final TransactionLoader transactionLoader;

    public TransactionService(TransactionLoader transactionLoader) {
        this.transactionLoader = transactionLoader;
    }

    public List<Transaction> getAllTransactions() {
        return transactionLoader.getAllTransactions();
    }

    public Transaction getTransactionById(String transactionId) {
        return transactionLoader.getTransactionById(transactionId)
            .orElseThrow(() -> new ResourceNotFoundException("Transaction", transactionId));
    }

    public List<Transaction> getTransactionsForAccount(String accountId) {
        return transactionLoader.getTransactionsForAccount(accountId);
    }

    public List<Transaction> getTransactionsByStatus(String status) {
        return transactionLoader.getTransactionsByStatus(status);
    }

    public List<Transaction> getHighRiskTransactions(int minRiskScore) {
        return transactionLoader.getHighRiskTransactions(minRiskScore);
    }

    public List<Transaction> getPendingReviewTransactions() {
        return transactionLoader.getTransactionsByStatus("PENDING_REVIEW");
    }

    public List<Transaction> getBlockedTransactions() {
        return transactionLoader.getTransactionsByStatus("BLOCKED");
    }
}

File: frontend/src/app/core/services/transaction.service.ts
import { Injectable, inject } from '@angular/core';
import { Observable, BehaviorSubject } from 'rxjs';
import { tap } from 'rxjs/operators';
import { ApiService } from './api.service';
import { Transaction, TransactionEdge } from '@models/index';

@Injectable({
  providedIn: 'root'
})
export class TransactionService {
  private apiService = inject(ApiService);
  
  private transactionsSubject = new BehaviorSubject<Transaction[]>([]);
  private edgesSubject = new BehaviorSubject<TransactionEdge[]>([]);

  transactions$ = this.transactionsSubject.asObservable();
  edges$ = this.edgesSubject.asObservable();

  getTransactions(): Observable<Transaction[]> {
    return this.apiService.get<Transaction[]>('/api/transactions').pipe(
      tap(transactions => this.transactionsSubject.next(transactions))
    );
  }

  getTransactionById(transactionId: string): Observable<Transaction> {
    return this.apiService.get<Transaction>(`/api/transactions/${transactionId}`);
  }

  getTransactionsByAccount(accountId: string): Observable<Transaction[]> {
    return this.apiService.get<Transaction[]>(`/api/transactions?accountId=${accountId}`);
  }

  getHighRiskTransactions(minScore: number = 70): Observable<Transaction[]> {
    return this.apiService.get<Transaction[]>(`/api/transactions/high-risk?minScore=${minScore}`);
  }

  getPendingReviewTransactions(): Observable<Transaction[]> {
    return this.apiService.get<Transaction[]>('/api/transactions?status=PENDING_REVIEW');
  }

  getBlockedTransactions(): Observable<Transaction[]> {
    return this.apiService.get<Transaction[]>('/api/transactions?status=BLOCKED');
  }

  getTransactionEdges(): Observable<TransactionEdge[]> {
    return this.apiService.get<TransactionEdge[]>('/api/transaction-edges').pipe(
      tap(edges => this.edgesSubject.next(edges))
    );
  }

  getFlaggedEdges(): Observable<TransactionEdge[]> {
    return this.apiService.get<TransactionEdge[]>('/api/transaction-edges?flagged=true');
  }
}
